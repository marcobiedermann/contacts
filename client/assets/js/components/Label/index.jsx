import PropTypes from 'prop-types';
import React from 'react';
import styles from './style.css';

const Label = props => {
  const { children, htmlFor, ...otherProps } = props;

  return (
    <label className={styles.label} htmlFor={htmlFor} {...otherProps}>
      {children}
    </label>
  );
};

Label.propTypes = {
  children: PropTypes.node,
  htmlFor: PropTypes.string.isRequired,
};

Label.defaultProps = {
  children: null,
};

export default Label;
