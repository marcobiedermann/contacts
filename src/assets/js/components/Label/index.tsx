/* eslint-disable jsx-a11y/label-has-associated-control */

import React, { FC, LabelHTMLAttributes } from 'react';
import styles from './style.module.css';

export type LabelProps = LabelHTMLAttributes<HTMLLabelElement>;

const Label: FC<LabelProps> = (props) => <label className={styles.label} {...props} />;

export default Label;
