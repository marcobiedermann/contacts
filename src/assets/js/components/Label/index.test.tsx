import React from 'react';
import renderer from 'react-test-renderer';
import Label from '.';

describe('label component', () => {
  it('renders correctly', () => {
    expect.assertions(1);

    const tree = renderer.create(<Label htmlFor="input" />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
