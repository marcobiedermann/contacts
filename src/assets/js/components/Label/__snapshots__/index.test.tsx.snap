// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`label component should render correctly 1`] = `
<DocumentFragment>
  <label
    class="label"
    for="input"
  >
    Label
  </label>
</DocumentFragment>
`;
